cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp

```diff
@@ -206,25 +206,30 @@ class knn_callback_distr {
         ONEDAL_ASSERT(inp_distances.get_dimension(0) == len);
         ONEDAL_ASSERT(inp_distances.get_dimension(1) == k_neighbors_);
 
-        auto current_min_resp_dest = part_responses_.get_col_slice(k_neighbors_, 2 * k_neighbors_)
-                                         .get_row_slice(first, last);
-
-        copy_current_resp_event =
-            pr::select_indexed(queue_, inp_indices, train_responses_, current_min_resp_dest, deps);
+        pr::ndview<res_t, 2> current_min_resp_dest;
+        pr::ndview<res_t, 2> min_resp_dest;
+        if (result_options_.test(result_options::responses)) {
+            current_min_resp_dest = part_responses_.get_col_slice(k_neighbors_, 2 * k_neighbors_)
+                                        .get_row_slice(first, last);
+            copy_current_resp_event = pr::select_indexed(queue_,
+                                                         inp_indices,
+                                                         train_responses_,
+                                                         current_min_resp_dest,
+                                                         deps);
+            min_resp_dest = intermediate_responses_.get_row_slice(first, last);
+        }
 
         const pr::ndshape<2> typical_blocking(last - first, 2 * k_neighbors_);
         auto select = selc_t(queue_, typical_blocking, k_neighbors_);
 
         auto min_dist_dest = distances_.get_row_slice(first, last);
         auto min_indc_dest = indices_.get_row_slice(first, last);
-        auto min_resp_dest = intermediate_responses_.get_row_slice(first, last);
 
         // add global offset value to input indices
         ONEDAL_ASSERT(global_index_offset_ != -1);
-        auto treat_event = pr::treat_indices(queue_,
-                                             inp_indices,
-                                             global_index_offset_,
-                                             { copy_current_resp_event });
+        bk::event_vector ndeps = deps;
```

Comment:
does it make sense to reuse deps with
`deps.push_back(copy_current_resp_event);`

Reply:
Update: this was causing failure, so it does make sense to keep as is